source/NuGet.Lucene.Web/Controllers/PackagesController.cs

```diff
@@ -90,12 +90,24 @@ public HttpResponseMessage DownloadPackage(string id, string version="")
                 return result;
             }
 
-            result = Request.CreateResponse(HttpStatusCode.OK);
+            if (Request.Headers.Range != null)
+            {
+                try
+                {
+                    HttpResponseMessage partialResponse = Request.CreateResponse(HttpStatusCode.PartialContent);
+                    partialResponse.Content = new ByteRangeStreamContent(package.GetStream(), Request.Headers.Range, new MediaTypeWithQualityHeaderValue("application/zip"));
+                    return partialResponse;
```

Comment:
I did consider it, but I'm pretty certain it wouldn't be valid. The ETag/NotModified for example has to take into account the range requested. Content Disposition (and even mime type) wouldn't be relevant or the same, compared to full request - multiple ranges, for example produce multipart response, see the [spec](http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.16).